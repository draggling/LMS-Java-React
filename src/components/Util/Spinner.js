'use strict';
import React from 'react';
import RingLoader from 'react-spinners/RingLoader';
const Spinner = (...key) => {
	let content = '';
	key && typeof (key[0] === 'number' || key[0] === 'string')
		?(content = (
				<div className="d-flex justify-content-center" key={key}>
					<RingLoader size={150} color={'#00ccff'} loading={true} />
				</div>
		))
		: (content = (
				<div className="d-flex justify-content-center">
					<RingLoader size={150} color={'#00ccff'} loading={true} />
				</div>
			));
	return content;
};
export default Spinner;
