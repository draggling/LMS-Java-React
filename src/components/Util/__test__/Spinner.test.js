import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Spinner from '../Spinner';

it('renders without crashing', () => {
	const { asFragment } = render(Spinner());
	expect(asFragment()).toMatchSnapshot();
});

it('renders without crashing with key', () => {
	const { asFragment } = render(Spinner(111));
	expect(asFragment()).toMatchSnapshot();
});
