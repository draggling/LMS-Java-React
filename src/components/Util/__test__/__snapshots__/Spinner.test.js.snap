// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center"
    data-testid="spinner-with-key"
  >
    <div
      class="css-dyz6n"
    >
      <div
        class="css-kpuief"
      />
      <div
        class="css-1lkem5z"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without crashing with key 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center"
    data-testid="spinner-with-key"
  >
    <div
      class="css-dyz6n"
    >
      <div
        class="css-kpuief"
      />
      <div
        class="css-1lkem5z"
      />
    </div>
  </div>
</DocumentFragment>
`;
